pkg/logger/logger.go

```diff
@@ -43,8 +44,8 @@ type ZapLogger struct {
 
 var _ Logger = (*ZapLogger)(nil)
 
-func (l *ZapLogger) With(fields ...zap.Field) {
-	l.Logger = l.Logger.With(fields...)
+func (l *ZapLogger) With(fields ...zap.Field) Logger {
```

Comment:
I can't find any usages of this function that is not unit tests. What is the value of this change?

Reply:
The value of the change is that it fixes the behavior. The potential value of this change is that we can embed per request data into logs like I am aiming in https://github.com/openfga/openfga-envoy/pull/17

Reply:
BTW it is being used here https://github.com/openfga/openfga/blob/main/pkg/logger/logger_test.go#L111

Reply:
Check that it matches with the original zap logger interface - https://github.com/uber-go/zap/blob/d6b2d14fc541ed17b31b794afc139f03648842ca/logger.go#L180

Reply:
OK, understood now, the goal is to be able to do this:

```go
reqID := req.Attributes.GetRequest().GetHttp().GetHeaders()["x-request-id"]
logger := logger.With("request_id", reqID)
// ...
if err != nil {
		logger.Error("some error")
}
```